manual/src/cmds/intf-c.etex

```diff
@@ -2403,66 +2410,67 @@ section.
+known to the OCaml runtime system.  Threads created from OCaml (through
 the "Thread.create" function of the system threads library) are
-automatically known to the run-time system.  If the application
+automatically known to the runtime system.  If the application
 creates additional threads from C and wishes to callback into OCaml
-code from these threads, it must first register them with the run-time
+code from these threads, it must first register them with the runtime
 system.  The following functions are declared in the include file
 "<caml/threads.h>".
 
 \begin{itemize}
 \item
 "caml_c_thread_register()" registers the calling thread with the OCaml
-run-time system.  Returns 1 on success, 0 on error.  Registering an
+runtime system.  Returns 1 on success, 0 on error.  Registering an
 already-registered thread does nothing and returns 0.
 \item
 "caml_c_thread_unregister()"  must be called before the thread
-  terminates, to unregister it from the OCaml run-time system.
+  terminates, to unregister it from the OCaml runtime system.
 Returns 1 on success, 0 on error.  If the calling thread was not
 previously registered, does nothing and returns 0.
 \end{itemize}
 
-\subsection{ss:parallel-execution-long-running-c-code}{Parallel execution of long-running C code}
+\subsection{ss:parallel-execution-long-running-c-code}{Parallel
+execution of long-running C code and blocking system calls}
 
-The OCaml run-time system is not reentrant: at any time, at most one
-thread can be executing OCaml code or C code that uses the OCaml
-run-time system.  Technically, this is enforced by a ``master lock''
-that any thread must hold while executing such code.
+The OCaml runtime system is not reentrant: at any time, at most one
+thread on any domain can be executing OCaml code, or C code that uses
+the OCaml runtime system. Technically, this is enforced by a ``domain
+lock'' that any thread must hold while executing such code.
 
-When OCaml calls the C code implementing a primitive, the master lock
+When OCaml calls the C code implementing a primitive, the domain lock
 is held, therefore the C code has full access to the facilities of the
-run-time system.  However, no other thread can execute OCaml code
-concurrently with the C code of the primitive.
+runtime system.  However, no other thread can execute OCaml code
+concurrently with the C code of the primitive on this domain.
 
 If a C primitive runs for a long time or performs potentially blocking
-input-output operations, it can explicitly release the master lock,
-enabling other OCaml threads to run concurrently with its operations.
-The C code must re-acquire the master lock before returning to OCaml.
-This is achieved with the following functions, declared in
-the include file "<caml/threads.h>".
+input-output operations, it can explicitly release the domain lock,
+enabling other threads to run concurrently with its operations. The C
```

Comment:
I think this needs clarifying - releasing the domain lock allows other systhreads attached to the domain to execute OCaml code, but systhreads attached to other domains are already able to execute in parallel. It's probably a good opportunity to start using "parallel" vs "concurrent" (domains/effects confusion, etc.)

Reply:
Well yes, that's why it says "concurrently".